File: kattis/friday_the_13th.pl
% https://open.kattis.com/problems/friday
% 1s
% 2.0 Easy

% All year start on a Sunday (0)
% If "January" have at least 13 days, then it's a Friday.
main :-
    readln([_|S],end_of_file),
    s(S).
s([]).
s([_D,M|Ss]) :-
    length(Ms,M),
    append(Ms,Rs,Ss),
    f(Ms,0,0,F),
    writeln(F),
    s(Rs).
f([],_,F,F).
f([M|Ms],D0,F0,F) :-
    (M < 13 ->
        F1 is F0
    ;
        D13 is (D0 + 13-1) mod 7,
        (D13 =:= 5 ->
            
            F1 is F0+1
        ;
            F1 is F0
        )
    ),
    D1 is (D0+M) mod 7,
    f(Ms,D1,F1,F).


/*
% Compressed: 248 chars. Not short enough for Top 10 (172..210 chars)
main:-readln([_|S],end_of_file),s(S).
s([]). s([_D,M|Ss]):-length(Ms,M),append(Ms,Rs,Ss),f(Ms,0,0,F),writeln(F),s(Rs).
f([],_,F,F). f([M|Ms],D0,F0,F):-
(M<13->F1 is F0;T is(D0+13-1)mod 7,(T=:=5->F1 is F0+1;F1 is F0)),D1 is(D0+M)mod 7,f(Ms,D1,F1,F).
*/

File: kattis/friday_the_13th.py
# https://open.kattis.com/problems/friday
# 1s
# 2.0 Easy

# It's correct
"""
import sys
v=sys.stdin.readlines()[1::]
for i in range(len(v)//2):
    s = [int(x) for x in v[2*i+1].split()]
    l=len(s)
    f = 0
    d = 0
    for m in s:
        if m >= 13:
            if (d + 13-1) % 7 == 5:
                f+=1
        d = (d+m)%7
    print(f)
"""

# And now let's golf it: 179 chars. Top 10, place 2 (it should be place 3)
import sys
v=sys.stdin.readlines()[1::]
for i in range(len(v)//2):
 s=[int(x)for x in v[2*i+1].split()];f=0;d=0
 for m in s:
  if m>=13 and(d+13-1)%7==5:f+=1
  d=(d+m)%7
 print(f)
